bin/live/pycbc_live_single_trigger_fits

```diff
@@ -273,30 +282,96 @@ logging.info('Sorting events into template duration bins')
 n_bins = duration_bin_edges.size - 1
 alphas = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
 counts = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
-binned_sngl_stats = {ifo: {} for ifo in args.ifos}
+event_bins = {}
+times_to_prune = {ifo: [] for ifo in args.ifos}
+
+# Note: cluster_over_time has quite a lot of logigng, so turn
+# logging off temporarily
+logger = logging.getLogger()
+# Get current logging level
+level = logger.level
+logger.setLevel(logging.WARN)
 
 for ifo in events.keys():
     # Sort the events into their bins
-    event_bin = np.array([tbins[d]
-                          for d in events[ifo].data['template_duration']])
+    event_bins[ifo] = np.array([tbins[d]
+                               for d in events[ifo].data['template_duration']])
+
+    if args.prune_loudest:
+        for bin_num in range(n_bins):
+            inbin = event_bins[ifo] == bin_num
+
+            binned_events = events[ifo].data[args.sngl_ranking][inbin]
+            binned_event_times = events[ifo].data['end_time'][inbin]
+
+            # Cluster the triggers in time (use 2x the pruning window)
+            # to make sure we aren't just getting the same event N times
+            # in different templates within the same bin
+            cidx = cluster_over_time(binned_events, binned_event_times,
+                                     args.prune_window * 2)
+
+            # Deal with the case where there are fewer clustered triggers
+            # than we would prune
+            if args.prune_loudest > cidx.size:
+                # Prune all triggers
```

Comment:
suggest 'There are fewer triggers than the number specified, so prune them all'

but also .. won't the result of this be a nonsensical fit?

Reply:
Yes it would, the issue came up mostly in the testing using fewer triggers so this didn't work, so it needed a workaround

This is then taken into consideration later on the `if not np.count_nonzero(inbin)` line